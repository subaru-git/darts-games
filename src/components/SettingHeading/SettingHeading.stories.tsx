/* eslint-disable import/named */
import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import SettingHeading from './SettingHeading';

export default {
  title: 'Components/SettingHeading',
  component: SettingHeading,
} as ComponentMeta<typeof SettingHeading>;

const Template: ComponentStory<typeof SettingHeading> = (args) => <SettingHeading {...args} />;

export const Default = Template.bind({});
Default.args = {};
