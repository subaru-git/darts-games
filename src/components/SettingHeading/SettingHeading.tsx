import React, { FC } from 'react';
import {
  Flex,
  Text,
  IconButton,
  Popover,
  PopoverArrow,
  PopoverBody,
  PopoverCloseButton,
  PopoverContent,
  PopoverHeader,
  PopoverTrigger,
} from '@chakra-ui/react';
import { IoIosHelpCircleOutline } from 'react-icons/io';

type SettingHeadingProps = {
  title: string;
  hintHeader: string;
  hintBody: string;
};

const SettingHeading: FC<SettingHeadingProps> = ({ title, hintHeader, hintBody }) => {
  return (
    <Flex alignItems={'center'}>
      <Text fontSize={'sm'} fontWeight={'bold'} color={'gray.500'}>
        {title}
      </Text>
      <Popover>
        <PopoverTrigger>
          <IconButton
            icon={<IoIosHelpCircleOutline />}
            variant={'ghost'}
            color={'gray.500'}
            aria-label={`${title} help`}
          />
        </PopoverTrigger>
        <PopoverContent>
          <PopoverArrow />
          <PopoverCloseButton />
          <PopoverHeader border={'0'}>{hintHeader}</PopoverHeader>
          <PopoverBody>{hintBody}</PopoverBody>
        </PopoverContent>
      </Popover>
    </Flex>
  );
};

export default SettingHeading;
