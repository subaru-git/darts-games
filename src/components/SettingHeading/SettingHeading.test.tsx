import { render } from '@testing-library/react';
import SettingHeading from './SettingHeading';

test('should rendering', () => {
  const { container } = render(<SettingHeading />);
  expect(container).toMatchSnapshot();
});
