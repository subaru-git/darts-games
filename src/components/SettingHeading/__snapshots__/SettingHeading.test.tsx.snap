// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <span
      class="text-sm font-bold text-gray-500"
    >
      title
    </span>
    <button
      aria-controls="popover-content-:r1:"
      aria-expanded="false"
      aria-haspopup="dialog"
      aria-label="title help"
      class="chakra-button css-7u5se0"
      id="popover-trigger-:r1:"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 76c48.1 0 93.3 18.7 127.3 52.7S436 207.9 436 256s-18.7 93.3-52.7 127.3S304.1 436 256 436c-48.1 0-93.3-18.7-127.3-52.7S76 304.1 76 256s18.7-93.3 52.7-127.3S207.9 76 256 76m0-28C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
        />
        <path
          d="M256.7 160c37.5 0 63.3 20.8 63.3 50.7 0 19.8-9.6 33.5-28.1 44.4-17.4 10.1-23.3 17.5-23.3 30.3v7.9h-34.7l-.3-8.6c-1.7-20.6 5.5-33.4 23.6-44 16.9-10.1 24-16.5 24-28.9s-12-21.5-26.9-21.5c-15.1 0-26 9.8-26.8 24.6H192c.7-32.2 24.5-54.9 64.7-54.9zm-26.3 171.4c0-11.5 9.6-20.6 21.4-20.6 11.9 0 21.5 9 21.5 20.6s-9.6 20.6-21.5 20.6-21.4-9-21.4-20.6z"
        />
      </svg>
    </button>
    <div
      class="chakra-popover__popper css-0"
      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
    >
      <section
        aria-describedby="popover-body-:r1:"
        aria-labelledby="popover-header-:r1:"
        class="chakra-popover__content css-1ab1jlb"
        id="popover-content-:r1:"
        role="dialog"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
        tabindex="-1"
      >
        <div
          class="chakra-popover__arrow-positioner css-0"
          data-popper-arrow=""
          style="position: absolute;"
        >
          <div
            class="chakra-popover__arrow css-0"
            data-popper-arrow-inner=""
          />
        </div>
        <button
          aria-label="Close"
          class="chakra-popover__close-btn css-6sdaxy"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
        <header
          class="chakra-popover__header css-107tt68"
          id="popover-header-:r1:"
        >
          hint
        </header>
        <div
          class="chakra-popover__body css-0"
          id="popover-body-:r1:"
        >
          this is hint.
        </div>
      </section>
    </div>
  </div>
</div>
`;
